designer/apps/api/v1/views.py

```diff
@@ -43,7 +43,7 @@ def get_queryset(self):
         if len(models) == 1:
             queryset = models[0].objects.all()
         else:
-            queryset = Page.objects.all()
+            queryset = Page.objects.select_related('site').all()
```

Comment:
What does this do? Collect site at queryset time instead of serializer time?

Reply:
It reduces the number of queries required to get the data when the queryset is retrieved by keeping the duplicate queries to a minimum. 

In this case, each call to the serializer would then need to call get on the Site model to get the site.  By adding the select related here we do not need to query again.  

Recent work on Discovery to improve the API performance has made me a little overprotective query count.  I will double-check I am using this correctly.

Reply:
Ahh, that makes sense. One query gathering them all rather than each Program getting it's own. Cool cool :+1:

Reply:
I double-checked and this is the correct use of this method.  Here are the docs also for reference, https://docs.djangoproject.com/en/dev/ref/models/querysets/#select-related

Lastly by adding this the number of queries for the pages API goes from 16 to 12 on my localhost.